src/main/routes/metrics-route.ts

```diff
@@ -1,72 +1,73 @@
 import { LensApiRequest } from "../router"
 import { LensApi } from "../lens-api"
-import { PrometheusClusterQuery, PrometheusIngressQuery, PrometheusNodeQuery, PrometheusPodQuery, PrometheusProvider, PrometheusPvcQuery, PrometheusQueryOpts } from "../prometheus/provider-registry"
+import { Cluster } from "../cluster"
+import _ from "lodash"
 
 export type IMetricsQuery = string | string[] | {
   [metricName: string]: string;
 }
 
+// This is used for backoff retry tracking.
+const MAX_ATTEMPTS = 5
+const ATTEMPTS = [...(_.fill(Array(MAX_ATTEMPTS - 1), false)), true]
```

Comment:
If you want async retry, this is a good package, but no need to use here, just FYI https://github.com/vercel/async-retry

Reply:
Good point, if we need to use it again then I think I will pull in that package.